### PR TITLE
add shelf life transaction

### PR DESCRIPTION
**Pull request by bogdan34**

### CODE REVIEW

File: config/default/constants.ts [hunk 1]
```diff
@@ -68,6 +68,7 @@ const constants: IConstants = {
         CURRENT_BLOCK_VERSION: 1,
     },
     TRANSACTION_QUEUE_EXPIRE: 300,
+    TRANSACTION_SHELF_LIFE: 5 * 24 * 60 * 60,
```
Comment (macaqueee) [inline config/default/constants.ts]: maybe rename to  `TRANSACTION_TTL` or `TRANSACTION_KEEPALIVE`?

File: core/service/transaction.ts [hunk 2]
```diff
@@ -464,6 +464,15 @@ class TransactionService<T extends IAsset> implements ITransactionService<T> {
             return new ResponseEntity<void>({ errors: ['Invalid transaction timestamp. CreatedAt is in the future'] });
         }
 
+        if (SlotService.getSlotNumber(trs.createdAt + config.CONSTANTS.TRANSACTION_SHELF_LIFE)
+            < SlotService.getSlotNumber()
```
Comment (Verbalist) [inline core/service/transaction.ts]: < if trs.blockId ? SlotService.getSlotNumber(BlockRepo.getById(trs.blockId).createdAt): SlotService.getSlotNumber()

File: config/test/constants.ts [hunk 3]
```diff
@@ -68,6 +68,7 @@ const constants: IConstants = {
         CURRENT_BLOCK_VERSION: 1,
     },
     TRANSACTION_QUEUE_EXPIRE: 300,
+    TRANSACTION_SHELF_LIFE: 5 * 24 * 60 * 60,
```
Comment (macaqueee) [inline config/test/constants.ts]: increase this parameter. for test purposes we need more

File: config/default/constants.ts [hunk 4]
```diff
@@ -98,6 +99,7 @@ const constants: IConstants = {
     START_FEATURE_BLOCK: {
         ACCOUNT_BLACKLIST: 0,
         VOTE_WITH_ACTIVE_STAKE: 0,
+        FIX_TRANSACTION_IN_THE_PAST_BLOCK_HEIGHT: 455155,
```
Comment (axelsheva) [inline config/default/constants.ts]: Only on mainnet, for other configs set 0

**Pull request merged**